src/all/mangapark/src/eu/kanade/tachiyomi/extension/all/mangapark/MangaPark.kt

```diff
@@ -264,7 +304,7 @@ class MangaPark(
 
                 latch.countDown()
             } else {
-                latch.await()
+                latch.await(10, TimeUnit.SECONDS)
```

Comment:
Will this emit some relevant exception/toast for the user? If not- consider explicitly telling the user.